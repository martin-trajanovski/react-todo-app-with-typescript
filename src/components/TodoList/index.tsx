import React from 'react';
import { useSelector } from 'react-redux';
import TodoItem from '../TodoItem';
import { Todo } from 'src/interfaces';
import { AppState } from 'src/reducers';

const TodoList: React.FC = () => {
  const todos = useSelector((state: AppState) => state.todos);

  return (
    <div className="todo-list">
      {todos.map((todo: Todo) => (
        <TodoItem key={todo.id} todo={todo} />
      ))}
    </div>
  );
};

export default TodoList;
