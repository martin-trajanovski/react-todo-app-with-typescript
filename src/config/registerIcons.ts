import { library } from '@fortawesome/fontawesome-svg-core';
import { faTrash } from '@fortawesome/free-solid-svg-icons';

export default function registerIcons(): void {
  library.add(faTrash);
}
