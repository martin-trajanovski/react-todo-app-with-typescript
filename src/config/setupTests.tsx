// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { createStore, Store } from 'redux';
import { Provider } from 'react-redux';

import { rootReducer, AppState } from '../reducers';
import registerIcons from './registerIcons';

registerIcons();

export const initialTestingAppState: AppState = {
  todos: [{ id: 'test', text: 'Test todo app', completed: false }],
};

type WithReduxConfig = {
  initialState?: AppState;
  store?: Store;
};

interface RenderWithReduxResult extends RenderResult {
  store: Store<AppState>;
}

export const renderWithRedux = (
  ui: React.ReactElement,
  {
    initialState = initialTestingAppState,
    store = createStore(rootReducer, initialState),
  }: WithReduxConfig = {}
): RenderWithReduxResult => {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
};
