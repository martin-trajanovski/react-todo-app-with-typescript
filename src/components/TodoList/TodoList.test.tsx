import React from 'react';
import TodoList from '.';
import { renderWithRedux } from '../../config/setupTests';

test('should display todos correctly', () => {
  const { container } = renderWithRedux(<TodoList />);
  const todoList = container.querySelectorAll('.todo-item');

  expect(todoList.length).toBe(1);
});

test('should render todos correctly', () => {
  const { asFragment } = renderWithRedux(<TodoList />);

  expect(asFragment()).toMatchSnapshot();
});
