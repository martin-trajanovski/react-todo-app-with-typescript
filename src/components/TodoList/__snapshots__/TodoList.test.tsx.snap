// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render todos correctly 1`] = `
<DocumentFragment>
  <div
    class="todo-list"
  >
    <div
      class="todo-item "
    >
      <span
        class="todo-item-text"
      >
        Test todo app
      </span>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash fa-w-14 float-right"
        color="#dc3545"
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
